app/init.php

```diff
@@ -1139,20 +1131,30 @@ function (mixed $value) {
     $database = new Database($dbAdapter, $cache);
 
     $database
-        ->setNamespace('_' . $project->getInternalId())
         ->setMetadata('host', \gethostname())
         ->setMetadata('project', $project->getId())
         ->setTimeout(APP_DATABASE_TIMEOUT_MILLISECONDS);
 
+    if ($project->getAttribute('database') === DATABASE_SHARED_TABLES) {
+        $database
+            ->setShareTables(true)
+            ->setTenant($project->getInternalId())
+            ->setNamespace('');
+    } else {
+        $database
+            ->setShareTables(false)
+            ->setTenant(null)
+            ->setNamespace('_' . $project->getInternalId());
+    }
+
```

Comment:
@abnegate the value of `project->getAttribute('database')` is currently one of `db_fra1_v14x_02`, `db_fra1_v14x_03`, `db_fra1_v14x_04` etc. This allows us to allow new projects to be created on a new DB based on our needs and also gives us flexibility to route new projects to a new database in case there is some sort of a corruption. 

However with the introduction of the variable `DATABASE_SHARED_TABLES`, the `database` attribute can only be set to one specific database name which is (database_db_self_hosted_shared_tables). Which means we can only set it to one database. 

This can be problematic as we would be forced to use this one database forever?

Reply:
The `DATABASE_SHARED_TABLES` const will be temporary until we move all projects to use shared tables